### DISCUSSION TITLE
Reduce MQTT traffic

### DISCUSSION THREAD

**Discussion by pmknowles** [tony-fav/tasmota-blerry#85] [Q&A]
Is it possible to reduce the MQTT traffic? Once we know that it's working do we need to see the BLEDetails as well? The Tasmota command BLEDetails0 stops everything working so I guess you need the data from that. . Otherwise it's a brilliant piece of software. The log below is just over 1 minute with 2 devices. At the moment I'm using
`Rule2 on time#minute|2 do backlog bledetails2 A4C138EDEF74; delay 150; bledetails2 A4C1385C6ED5; delay 150; bledetails0 endon` which reduces the load a bit.

> 10:55:09.810 MQT: tele/HeltecBLE/BLE = {"DetailsBLE":{"mac":"A4C1385C6ED5","a":"BathroomTH","RSSI":-82,"p":"0D09475648353037355F36454435030388EC02010509FF88EC000202F86400"}}
10:55:09.920 MQT: tele/HeltecBLEdata/BathroomTH = {"Model":"GVH5075","Battery":100,"RSSI":-82,"Alias":"BathroomTH","Humidity":83.2,"Temperature":13.19,"Time":"2024-04-30T10:55:09","DewPoint":10.39,"MAC":"A4C1385C6ED5","Name":"BathroomTH"}
10:55:20.015 MQT: tele/HeltecBLE/BLE = {"DetailsBLE":{"mac":"A4C1385C6ED5","a":"BathroomTH","RSSI":-82,"p":"0D09475648353037355F36454435030388EC02010509FF88EC000202F76400"}}
10:55:20.926 MQT: tele/HeltecBLEdata/BathroomTH = {"Model":"GVH5075","Battery":100,"RSSI":-82,"Alias":"BathroomTH","Humidity":83.1,"Temperature":13.19,"Time":"2024-04-30T10:55:20","DewPoint":10.38,"MAC":"A4C1385C6ED5","Name":"BathroomTH"}
10:55:27.620 MQT: tele/HeltecBLE/BLE = {"DetailsBLE":{"mac":"A4C138EDEF74","a":"LandingTH","RSSI":-75,"p":"0D09475648353037355F45463734030388EC02010509FF88EC0001FEA16400"}}
10:55:27.932 MQT: tele/HeltecBLEdata/LandingTH = {"Model":"GVH5075","Battery":100,"RSSI":-75,"Alias":"LandingTH","Humidity":72.1,"Temperature":13.08,"Time":"2024-04-30T10:55:27","DewPoint":8.15,"MAC":"A4C138EDEF74","Name":"LandingTH"}
10:55:33.722 MQT: tele/HeltecBLE/BLE = {"DetailsBLE":{"mac":"A4C138EDEF74","a":"LandingTH","RSSI":-72,"p":"0D09475648353037355F45463734030388EC02010509FF88EC0001FEA16400"}}
10:55:41.874 MQT: tele/HeltecBLE/BLE = {"DetailsBLE":{"mac":"A4C138EDEF74","a":"LandingTH","RSSI":-72,"p":"0D09475648353037355F45463734030388EC02010509FF88EC0001FEA16400"}}
10:55:45.929 MQT: tele/HeltecBLE/BLE = {"DetailsBLE":{"mac":"A4C138EDEF74","a":"LandingTH","RSSI":-75,"p":"0D09475648353037355F45463734030388EC02010509FF88EC0001FEA16400"}}
10:55:58.189 MQT: tele/HeltecBLE/BLE = {"DetailsBLE":{"mac":"A4C138EDEF74","a":"LandingTH","RSSI":-72,"p":"0D09475648353037355F45463734030388EC02010509FF88EC0001FEA16400"}}
10:56:08.397 MQT: tele/HeltecBLE/BLE = {"DetailsBLE":{"mac":"A4C138EDEF74","a":"LandingTH","RSSI":-72,"p":"0D09475648353037355F45463734030388EC02010509FF88EC0001FEA06400"}}
10:56:08.956 MQT: tele/HeltecBLEdata/LandingTH = {"Model":"GVH5075","Battery":100,"RSSI":-72,"Alias":"LandingTH","Humidity":72,"Temperature":13.08,"Time":"2024-04-30T10:56:08","DewPoint":8.13,"MAC":"A4C138EDEF74","Name":"LandingTH"}
10:56:10.403 MQT: tele/HeltecBLE/BLE = {"DetailsBLE":{"mac":"A4C138EDEF74","a":"LandingTH","RSSI":-73,"p":"0D09475648353037355F45463734030388EC02010509FF88EC0001FEA06400"}}
10:56:16.556 MQT: tele/HeltecBLE/BLE = {"DetailsBLE":{"mac":"A4C138EDEF74","a":"LandingTH","RSSI":-71,"p":"0D09475648353037355F45463734030388EC02010509FF88EC0001FEA06400"}}

**Comment by tony-fav** [comment]
Hi @pmknowles, blerry acts as essentially a very complex Tasmota rule. As such, it needs a message to trigger off of and receive data from. That is the DetailsBLE messages. It's not possible for blerry to function without these.

Is there really concern over the amount of traffic? It's a tiny tiny fraction of traffic compared to even the simplest of modern webpages.

**Comment by pmknowles** [comment]
It's because it's doing it every few seconds for a device which changes state over a long period of time
Without my rule, with 7 devices, I got 334 messages in 10 minutes (1 every 2 seconds)
With my rule (essentially limiting each device to 1 reading every 5 minutes) I get 40.
When my MQTT broker is running on a Raspberry Pi (along with ZigBee2MQTT and openHAB) then, yes, the amount of unnecessary traffic is important - every time a message arrives openHAB runs a rule to see if it needs to respond to the message.

________________________________

**Reply by tony-fav** [reply]
It seems as though your rule can handle your situation. The issue here is generality. Your devices seem "chatty." That DetailsBLE message occurs each time the Tasmota device hears the BLE device advertise something. Where some devices only broadcast a couple times a minute. Others, like scales or thermometers, only broadcast when in use.  So, a general solution for disabling the datafeed (via turning off DetailsBLE) isn't really possible.

I will say, blerry only sends additional sensor payloads when there is a change to the data, and it spaces out sending discovery payloads. But this was more in consideration of ESP load, not load on the server.
